.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+name: Test
+on:
+  push:
+    branches: [main]
+  pull_request:
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    name: Linux ${{ matrix.elixir }}, ${{ matrix.otp }})
+    strategy:
+      matrix:
+        elixir: ["1.17.3"]
+        otp: ["27.1.2"]
+    env:
+      MIX_ENV: test
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp }}
+          elixir-version: ${{ matrix.elixir }}
+      - name: Install dependencies
+        run: mix deps.get
+      - name: Compile and check warnings
+        run: mix compile --warnings-as-errors
+      - name: Check formatting
+        run: mix format --check-formatted
+      - name: Run tests
+        run: mix test
+
+  macos:
+    runs-on: macos-latest
+    name: macOS
+    env:
+      MIX_ENV: test
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install
+        run: |
+          brew update
+          brew install erlang elixir
```

Comment:
I'm not too familiar with brew, should we specify versions here like the linux build?

Reply:
It seems the Homebrew Erlang package supports specification of the major release version `erlang@27`, but the elixir package does not. Future enhancement could be to use a different package manager.

(erlef/setup-beam is only supported on Linux and Windows)